syntax = "proto3";

import "google/protobuf/empty.proto";

package hello.proto;

service Greeter {
    // Typical RPC usage
    rpc SayHello (HelloRequest) returns (HelloResponse);
    rpc GetAllTransactions (google.protobuf.Empty) returns (stream HelloTransaction);

    // Continuous client-streaming RPC
    rpc MaybeSayHello(HelloRequest) returns (google.protobuf.Empty);
    rpc GetTransactionUpdates (google.protobuf.Empty) returns (stream HelloTransaction);
}

message HelloRequest {
    string name = 1;
}

message HelloResponse {
    string message = 1;
}

message HelloTransaction {
    HelloRequest request = 1;
    HelloResponse response = 2;
}

